draft-ietf-privacypass-protocol.md

```diff
@@ -12,984 +12,423 @@ stand_alone: yes
-cInput = Generate(m, "")
+client_context = SetupVerifiableClient(0x0004, pkI)
 ~~~
 
-A server that is expecting some non-empty `commit_resp` to be passed
-must abort the protocol on receiving a request containing an empty
-`commit_resp` value.
-
-Note: currently, no ciphersuites are supported that support working with
-empty commitment messages.
+Here, 0x0004 is the two-octet identifier corresponding to
+the OPRF(P-384, SHA-384) ciphersuite in {{OPRF}}.
 
-## Redemption phase {#redemption-phase}
-
-The redemption phase allows the client to anonymously reauthenticate to
-the server, using data that it has received from a previous issuance
-phase.
+The Client then creates an issuance request message for a random value `nonce`
+using the input challenge and Issuer key identifier as follows:
 
 ~~~
-  Client(info)                                Server(skS, pkS)
-  ------------------------------------------------------------
-  token = store[server.id].pop()
-  req = Redeem(token, info)
-
-                               req
-                        ------------------>
-
-                               if (dsIdx.includes(req.data)) {
-                                 raise ERR_DOUBLE_SPEND
-                               }
-                               resp = Verify(pkS, skS, req)
-                               if (resp.success) {
-                                 dsIdx.push(req.data)
-                               }
-
-                                resp
-                        <------------------
-  Output resp
+nonce = random(32)
+context = SHA256(challenge)
+token_input = concat(0x0001, nonce, context, key_id)
+blind, blinded_message = client_context.Blind(nonce)
 ~~~
 
-### Client info {#client-info}
-
-The client input `info` is arbitrary byte data that is used for linking
-the redemption request to the specific session. We RECOMMEND that `info`
-is constructed as the following concatenated byte-encoded data:
+The Client then creates a TokenRequest structured as follows:
 
 ~~~
-len(aux) || aux || len(server.id) || server.id || current_time()
+struct {
+   uint8_t token_type;
+   uint8_t token_key_id;
+   uint8_t blinded_msg[Nk];
+} TokenRequest;
 ~~~
 
-where `len(x)` is the length of `x` in bytes, and `aux` is arbitrary
-auxiliary data chosen by the client. The usage of `current_time()`
-allows the server to check that the redemption request has happened in
-an appropriate time window.
-
-### Double-spend protection
+The structure fields are defined as follows:
 
-To protect against clients that attempt to spend a value `req.data` more
-than once, the server uses an index, `dsIdx`, to collect valid inputs it
-witnesses. Since this store needs to only be optimized for storage and
-querying, a structure such as a Bloom filter suffices. The storage
-should be parameterized to live as long as the server keypair that is in
-use. See {{sec-reqs}} for more details.
+- "version" is a 1-octet integer, which matches the version in the challenge.
```

Comment:
Should this be token_type or the struct updated up above?

Reply:
Yep!